### PR TITLE
Update color picker

### PR DESCRIPTION
**Pull request by joseartrivera**
### Description of the changes:
- Update color picker to use defined set of colors instead of the default color picker
- Don't reinitialize the graph when changing colors
- Some code and UI clean up

### How changes were validated:
- Manual testing

### CODE REVIEW

File: src/CalcViewModel/CalcViewModel.vcxproj.filters [hunk 1]
```diff
@@ -186,29 +175,26 @@
...
-      <Filter>Common\Automation</Filter>
-    </ClInclude>
-    <ClInclude Include="Common\Automation\LiveRegionHost.h">
-      <Filter>Common\Automation</Filter>
+    <ClInclude Include="Common\CalculatorButtonUser.h">
+      <Filter>Common</Filter>
     </ClInclude>
     <ClInclude Include="Common\Automation\NarratorAnnouncement.h">
-      <Filter>Common\Automation</Filter>
+      <Filter>Common%255cAutomation</Filter>
```
Comment (rudyhuyn) [inline src/CalcViewModel/CalcViewModel.vcxproj.filters]: This is weird!

File: src/Calculator/StylePanel.xaml.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+{
+    SelectColor(newColor);
+}
+
+void StylePanel::ColorChooserLoaded(Object ^ sender, RoutedEventArgs ^ e)
+{
+    SelectColor(SelectedColor);
+}
+
+void StylePanel::SelectColor(SolidColorBrush ^ selectedColor)
```
Comment (rudyhuyn) [inline src/Calculator/StylePanel.xaml.cpp]: You can also let GridView manage the selection indicator if you modify `GridView::ItemContainerStyle`.

Once done, the only thing you will need to do here will be (in c#, sorry!):   `ColorChooser.SelectedItem = ColorChooser.Items.Where(t => selectedColor.Color.Equals(((SolidColorBrush)t).Color)).FirstOrDefault();` 


```xml
        <Style TargetType="GridViewItem" x:Key="GridViewItemExpanded">
            <Setter Property="Background" Value="Transparent" />
            <Setter Property="TabNavigation" Value="Local" />
            <Setter Property="IsHoldingEnabled" Value="True" />
            <Setter Property="HorizontalContentAlignment" Value="Center" />
            <Setter Property="VerticalContentAlignment" Value="Center" />
            <Setter Property="Margin" Value="0,0,4,4" />
            <Setter Property="AllowDrop" Value="False" />
            <Setter Property="UseSystemFocusVisuals" Value="{StaticResource UseSystemFocusVisuals}" />
            <Setter Property="FocusVisualMargin" Value="-2" />
            <Setter Property="Template">
                <Setter.Value>
                    <ControlTemplate TargetType="GridViewItem">
                        <Grid x:Name="ContentBorder"
                        Control.IsTemplateFocusTarget="True"
                        FocusVisualMargin="{TemplateBinding FocusVisualMargin}">
                            <VisualStateManager.VisualStateGroups>
                                <VisualStateGroup x:Name="FocusStates">
                                    <VisualState x:Name="Focused">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Visibility" Value="Collapsed" />
                                        </VisualState.Setters>
                                    </VisualState>
                                    <VisualState x:Name="Unfocused" />
                                </VisualStateGroup>
                                <VisualStateGroup x:Name="CommonStates">
                                    <VisualState x:Name="Normal">
                                        <Storyboard>
                                            <PointerUpThemeAnimation Storyboard.TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="PointerOver">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Opacity" Value="1"/>
                                            <Setter Target="BorderRectangle.Stroke" Value="{ThemeResource SystemControlHighlightListLowBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightListLowBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerUpThemeAnimation Storyboard.TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="Pressed">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Opacity" Value="1"/>
                                            <Setter Target="BorderRectangle.Stroke" Value="{ThemeResource SystemControlHighlightListMediumBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightListMediumBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerDownThemeAnimation TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="Selected">
                                        <VisualState.Setters>
                                            <Setter Target="Ellipse.StrokeThickness" Value="3" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightAccentBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerUpThemeAnimation Storyboard.TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="PointerOverSelected">
                                        <VisualState.Setters>
                                            <Setter Target="Ellipse.StrokeThickness" Value="3" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightListAccentMediumBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerUpThemeAnimation Storyboard.TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                    <VisualState x:Name="PressedSelected">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Opacity" Value="1"/>
                                            <Setter Target="BorderRectangle.Stroke" Value="{ThemeResource SystemControlHighlightListAccentHighBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryBrush" Value="{ThemeResource SystemControlHighlightListAccentHighBrush}" />
                                            <Setter Target="ContentBorder.FocusVisualSecondaryThickness" Value="2" />
                                        </VisualState.Setters>
                                        <Storyboard>
                                            <PointerDownThemeAnimation TargetName="Ellipse" />
                                        </Storyboard>
                                    </VisualState>
                                </VisualStateGroup>
                                <VisualStateGroup x:Name="DisabledStates">
                                    <VisualState x:Name="Enabled" />
                                    <VisualState x:Name="Disabled">
                                        <VisualState.Setters>
                                            <Setter Target="BorderRectangle.Opacity" Value="1"/>
                                        </VisualState.Setters>
                                    </VisualState>
                                </VisualStateGroup>
                            </VisualStateManager.VisualStateGroups>
                            <Ellipse x:Name="Ellipse"
                            HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}"
                            VerticalAlignment="{TemplateBinding VerticalContentAlignment}"
                                     Fill="{TemplateBinding Content}" Height="32" Width="32"
                                     StrokeThickness="0" Stroke="{ThemeResource StylePanelSelectedBrush}"
                            Margin="{TemplateBinding Padding}" />
                            <Rectangle x:Name="BorderRectangle"
                            IsHitTestVisible="False"
                            Stroke="{ThemeResource SystemControlHighlightListAccentLowBrush}"
                            StrokeThickness="2"
                            Opacity="0" />
                        </Grid>
                    </ControlTemplate>
                </Setter.Value>
            </Setter>
        </Style>
```

File: src/Calculator/App.xaml [hunk 3]
```diff
@@ -46,8 +46,29 @@
                                   FallbackColor="{ThemeResource SystemChromeMediumColor}"
                                   TintColor="{ThemeResource SystemChromeLowColor}"
                                   TintOpacity="0.7"/>
+                    <SolidColorBrush x:Key="EquationTextBoxTransparentBackgroundBrush" Color="Transparent"/>
                     <SolidColorBrush x:Key="EquationBoxPointerOverBackgroundBrush" Color="{ThemeResource SystemControlBackgroundAltMediumBrush}"/>
                     <SolidColorBrush x:Key="EquationBoxHoverButtonForegroundBrush" Color="{ThemeResource SystemBaseHighColor}"/>
+                    <SolidColorBrush x:Key="EquationBoxBorderBrush" Color="{ThemeResource TextControlBackground}"/>
```
Comment (DavidShoe) [inline src/Calculator/App.xaml]: Todo: Add high contrast colors mapping

File: src/Calculator/Views/GraphingCalculator/EquationInputArea.xaml [hunk 4]
```diff
@@ -3,80 +3,68 @@
...
+                                              Margin="0,0,3,0"
+                                              Style="{StaticResource EquationTextBoxStyle}"
+                                              EquationColor="{x:Bind LineColor, Mode=OneWay}"
+                                              GotFocus="InputTextBox_GotFocus"
+                                              KeyUp="InputTextBox_KeyUp"
+                                              LostFocus="InputTextBox_LostFocus"
+                                              RemoveButtonClicked="EquationTextBox_RemoveButtonClicked">
+                        <controls:EquationTextBox.KeyGraphFeaturesContent>
+                            <StackPanel x:Name="KeyGraphFeaturesGrid" Visibility="{x:Bind KeyGraphFeaturesVisibility, Mode=OneWay}">
+                                <TextBlock Text="Key Graph Features"/>
```
[the remark this answers is not in the document]
Comment (joseartrivera) [reply src/Calculator/Views/GraphingCalculator/EquationInputArea.xaml]: This is temporary UI that will be removed in an upcoming PR!

File: src/CalcViewModel/CalcViewModel.vcxproj.filters [hunk 5]
```diff
@@ -1,26 +1,12 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (sanderl) [inline src/CalcViewModel/CalcViewModel.vcxproj.filters]: You should probably change this back to 15 to be on the safe side.

Comment (joseartrivera) [reply src/CalcViewModel/CalcViewModel.vcxproj.filters]: fixed

**Pull request merged**